### PR TITLE
Fix multiple bugs

### PR DESCRIPTION
**Pull request by siriak**

### CODE REVIEW

File: Core/StringHelper.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            unchecked
+            {
+                var hash1 = 5381;
+                var hash2 = hash1;
+
+                for(var i = 0; i < str.Length && str[i] != '\0'; i += 2)
+                {
+                    hash1 = ((hash1 << 5) + hash1) ^ str[i];
+                    if (i == str.Length - 1 || str[i+1] == '\0')
```
Comment (siriak) [inline Core/StringHelper.cs:15]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- Core/DateTimeHelpers.cs  1
- Core/StringHelper.cs  5
         

Clones removed
==============
+ Core.Tests/TransactionTests.cs  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033&bid=19895607)

File: Core/StringHelper.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+                        break;
+                    hash2 = ((hash2 << 5) + hash2) ^ str[i+1];
+                }
+
+                return hash1 + (hash2*1566083941);
+            }
+        }
+    }
+}
```
Comment (siriak) [inline Core/StringHelper.cs:24]: Codacy found an issue: [Add a new line at the end of the file 'StringHelper.cs'.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033)

File: Core/DateTimeHelpers.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace Core
+{
+    public static class DateTimeHelpers
+    {
+        public static Date ToDate(this DateTime dt) => new Date(dt);
+    }
+}
```
Comment (siriak) [inline Core/DateTimeHelpers.cs:9]: Codacy found an issue: [Add a new line at the end of the file 'DateTimeHelpers.cs'.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033)

File: Core/StringHelper.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+                for(var i = 0; i < str.Length && str[i] != '\0'; i += 2)
+                {
+                    hash1 = ((hash1 << 5) + hash1) ^ str[i];
+                    if (i == str.Length - 1 || str[i+1] == '\0')
+                        break;
+                    hash2 = ((hash2 << 5) + hash2) ^ str[i+1];
+                }
+
+                return hash1 + (hash2*1566083941);
```
Comment (siriak) [inline Core/StringHelper.cs:20]: Codacy found an issue: [Add underscores to this numeric value for readability.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033)

File: Core/StringHelper.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            unchecked
+            {
+                var hash1 = 5381;
+                var hash2 = hash1;
+
+                for(var i = 0; i < str.Length && str[i] != '\0'; i += 2)
+                {
+                    hash1 = ((hash1 << 5) + hash1) ^ str[i];
+                    if (i == str.Length - 1 || str[i+1] == '\0')
+                        break;
```
Comment (siriak) [inline Core/StringHelper.cs:16]: Codacy found an issue: [Refactor the code in order to remove this break statement.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Core/DateTimeHelpers.cs  1
- Core/StringHelper.cs  5
         

Clones removed
==============
+ Core.Tests/TransactionTests.cs  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033&bid=19895607)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Core/StringHelper.cs  5
- Core/DateTimeHelpers.cs  1
         

Clones removed
==============
+ Core.Tests/TransactionTests.cs  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033&bid=19895607)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Core/DateTimeHelpers.cs  1
- Core/StringHelper.cs  5
         

Clones removed
==============
+ Core.Tests/TransactionTests.cs  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033&bid=19895607)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Core/HashHelper.cs  5
- Core/DateTimeHelpers.cs  1
         

Clones removed
==============
+ Core.Tests/TransactionTests.cs  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6077033&bid=19895607)

**Review by Adelhaide22 — APPROVED** [review]

**Pull request merged**